ci-deployment-cleanup/helm-chart/cleanup.sh

```diff
@@ -0,0 +1,365 @@
+    done
+else
+    echo "PR-based cleanup: disabled"
+fi
+
+if [ "$DRY_RUN" = "true" ]; then
+    echo "DRY RUN MODE: No actual deletions will be performed"
+fi
+
+debug_log "Initialization complete, starting namespace discovery"
+
+# Function to calculate age in seconds
+calculate_age() {
+    local timestamp="$1"
+    local current_time=$(date +%s)
+    
+    debug_log "Calculating age for timestamp: $timestamp"
+    
+    # Kubernetes timestamps are in ISO 8601 format, need to handle them properly
+    local creation_time
+    if command -v gdate >/dev/null 2>&1; then
+        # Use GNU date if available (Linux with coreutils)
+        creation_time=$(gdate -d "$timestamp" +%s 2>/dev/null || echo "0")
+        debug_log "Used gdate for timestamp parsing"
+    else
+        # For Alpine Linux/BusyBox date, we need to parse the ISO 8601 format manually
+        # Format: 2025-05-28T13:50:39Z
+        local year month day hour minute second
+        year=$(echo "$timestamp" | cut -d'-' -f1)
+        month=$(echo "$timestamp" | cut -d'-' -f2)
+        day=$(echo "$timestamp" | cut -d'T' -f1 | cut -d'-' -f3)
+        hour=$(echo "$timestamp" | cut -d'T' -f2 | cut -d':' -f1)
+        minute=$(echo "$timestamp" | cut -d':' -f2)
+        second=$(echo "$timestamp" | cut -d':' -f3 | sed 's/Z$//')
+        
+        debug_log "Parsed timestamp components: $year-$month-$day $hour:$minute:$second"
+        
+        # Use BusyBox date with explicit format
+        local formatted_timestamp="${year}-${month}-${day} ${hour}:${minute}:${second}"
+        creation_time=$(date -d "$formatted_timestamp" +%s 2>/dev/null || echo "0")
+        debug_log "Used BusyBox date for timestamp parsing"
+    fi
+    
+    if [ "$creation_time" = "0" ]; then
+        debug_log "Failed to parse timestamp, returning age 0"
+        echo "0"
+    else
+        local age=$((current_time - creation_time))
+        debug_log "Calculated age: $age seconds"
+        echo "$age"
+    fi
+}
+
+# Function to format age for display
+format_age() {
+    local age_seconds="$1"
+    local age_hours=$((age_seconds / 3600))
+    local age_days=$((age_hours / 24))
+    
+    if [ $age_days -gt 0 ]; then
+        echo "${age_days}d $((age_hours % 24))h"
+    else
+        echo "${age_hours}h"
+    fi
+}
+
+# Function to format hours to days+hours for thresholds
+format_hours_threshold() {
+    local hours="$1"
+    local days=$((hours / 24))
+    
+    if [ $days -gt 0 ]; then
+        echo "${days}d ($((hours % 24))h)"
+    else
+        echo "${hours}h"
+    fi
+}
+
+# Function to check if namespace matches any allowed pattern
+matches_pattern() {
+    local namespace="$1"
+    debug_log "Checking if namespace '$namespace' matches any allowed patterns"
+    
+    if [ "$ENFORCE_NAME_PATTERNS" = "true" ]; then
+        for pattern in $NAMESPACE_PATTERNS; do
+            debug_log "Testing pattern: $pattern"
+            if [[ "$namespace" =~ $pattern ]]; then
```

Comment:
this doesn't work for lookahead regex patterns like `^renku-ci-(?!cleanup).*` so we should make sure not to use those